src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Conversation.cshtml

```diff
@@ -11,35 +11,89 @@
+                                var divId = $"rev-{revision.Key.RevisionId}";
 
-                            <div class="card-header p-2 text-black-50 clickable" id="header-@revision.Key.RevisionId" data-toggle="collapse" data-target="#@divId">
+                                <div class="card-header p-2 text-black-50 clickable" id="header-@revision.Key.RevisionId" data-toggle="collapse" data-target="#@divId">
                                     @revision.Key.DisplayName
-                            </div>
-                            <div id=@divId class="collapse show" data-revision-id=@revision.Key.RevisionId aria-labelledby="header-@revision.Key.RevisionId">
-                                <table class="code-window">
-                                    @foreach (var thread in revision.Value)
-                                    {
-                                        var elementId = thread.LineId;
-                                        <tr class="code-line"><td class="code p-2" style="word-break: break-all;"><a class="comment-url" asp-page="Review"
+                                </div>
+                                <div id=@divId class="collapse show" data-revision-id=@revision.Key.RevisionId aria-labelledby="header-@revision.Key.RevisionId">
+                                    <table class="code-window">
+                                        @foreach (var thread in revision.Value)
+                                        {
+                                            var elementId = thread.LineId;
+                                            <tr class="code-line"><td class="code p-2" style="word-break: break-all;"><a class="comment-url" asp-page="Review"
                                                 asp-route-id=@Model.Review.ReviewId
                                                 asp-route-revisionId=@revision.Key.RevisionId
                                                 asp-fragment=@Uri.EscapeUriString(elementId)>@elementId</a></td></tr>
-                                        <partial name="_CommentThreadPartial" model="@thread" />
-                                    }
-                                </table>
-                            </div>
-                        }
-                    </div>
+                                            <partial name="_CommentThreadPartial" model="@thread" />
+                                        }
+                                    </table>
+                                </div>
+                            }
+                        </div>
+                    }
+                    @if (Model.UsageSampleThreads.Count > 0)
+                    {
+                        var skipped = 1;
+                        <br />
+                        <div class="border rounded">
+                            @foreach (var revision in Model.UsageSampleThreads.Reverse())
+                            {
+                                @if (revision.Key.RevisionIsDeleted) 
```

Comment:
Can we avoid fetching this from DB instead of filtering it out in UI side?

Reply:
Other Alternate approach is to do something like below:

@foreach (var revision in Model.UsageSampleThreads.Reverse().Where( x = > x.Key.RevisionIsDeleted == false))
{